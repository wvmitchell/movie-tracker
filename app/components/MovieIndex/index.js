import React, { Component } from 'react'
import _ from 'lodash'
import MovieCard from '../MovieCard'
import MoviesApi from '../../apis/moviesapi.js'
import { moviedbAPIKey } from '../../keys'

class MovieIndex extends Component {
  constructor() {
    super();
    this.moviesApi = new MoviesApi(moviedbAPIKey)

    this.state = {
      nowPlaying: [],
    }
  }

  componentDidMount = async () => { 
    const nowPlaying = await this.moviesApi.fetchNowPlaying()
    this.setState({ nowPlaying })
  }

  renderMovieCards = (movies) => (
    _.map(movies, (movie, index) => <MovieCard movie={movie} key={index} />)
  )

  render = () => (
    <div>
      { this.renderMovieCards(this.state.nowPlaying) }
    </div>
  )
}

export default MovieIndex
