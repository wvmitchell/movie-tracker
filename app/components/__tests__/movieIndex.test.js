import React from 'react'
import { shallow } from 'enzyme'
import MovieIndex from '../movieIndex'

describe('MovieIndex', () => {
  it('should match the snapshot', () => {
    const renderedMovieIndex = shallow(<MovieIndex />)
    expect(renderedMovieIndex).toMatchSnapshot()
  })
})
