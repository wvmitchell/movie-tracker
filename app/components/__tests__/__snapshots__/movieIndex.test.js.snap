// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieIndex should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieIndex />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "MOVIES",
    },
    "ref": null,
    "rendered": "MOVIES",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "MOVIES",
      },
      "ref": null,
      "rendered": "MOVIES",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
